src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -742,9 +742,11 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
 
                 DocumentsStorage.PutOperationResults? lastPutResult = null;
 
+                using IEnumerator<AttachmentStream> attachmentIterator = AttachmentStreams?.GetEnumerator();
+                
                 for (int i = ParsedCommands.Offset; i < ParsedCommands.Count; i++)
                 {
-                    var cmd = ParsedCommands.Array[ParsedCommands.Offset + i];
```

Comment:
It is not related to the main issue but I think it is wrong (It probably worked just because the offset was always zero)

Reply:
Yes, that is fine, because we start from `Offset`